model/host/host.go

```diff
@@ -1778,6 +1788,57 @@ func (h *Host) SetInstanceType(instanceType string) error {
 	return nil
 }
 
+func AggregateSpawnhostData() (*SpawnHostUsage, error) {
+	res := []SpawnHostUsage{}
+	pipeline := []bson.M{
+		{"$match": bson.M{
+			StartedByKey: bson.M{"$exists": true, "$ne": ""},
+			StatusKey:    bson.M{"$in": evergreen.UpHostStatus},
+		}},
+		{"$group": bson.M{
+			"_id":         nil,
+			"hosts":       bson.M{"$sum": 1},
+			"unexpirable": bson.M{"$sum": bson.M{"$cond": []interface{}{"$" + NoExpirationKey, 1, 0}}},
+			"users":       bson.M{"$addToSet": "$" + StartedByKey},
+		}},
+		{"$project": bson.M{
+			"_id":                     "0",
+			"total_hosts":             "$hosts",
+			"total_unexpirable_hosts": "$unexpirable",
+			"num_users_with_hosts":    bson.M{"$size": "$users"},
+		}},
+	}
+
+	if err := db.Aggregate(Collection, pipeline, &res); err != nil {
+		return nil, errors.Wrap(err, "error aggregating hosts")
+	}
+	if len(res) == 0 {
+		return nil, errors.New("no host results found")
+	}
+
+	pipeline = []bson.M{
+		{"$group": bson.M{
+			"_id":     nil,
+			"volumes": bson.M{"$sum": 1},
+			"size":    bson.M{"$sum": "$" + VolumeSizeKey},
+			"users":   bson.M{"$addToSet": "$" + VolumeCreatedByKey},
+		}},
+		{"$project": bson.M{
+			"_id":                    "0",
+			"total_volumes":          "$volumes",
+			"total_volume_size":      "$size",
+			"num_users_with_volumes": bson.M{"$size": "$users"},
+		}},
+	}
+	if err := db.Aggregate(VolumesCollection, pipeline, &res); err != nil {
+		return nil, errors.Wrap(err, "error aggregating volumes")
+	}
+	if len(res) == 0 {
```

Comment:
if there are no volumes could there still be hosts to report?

Reply:
len(res) here also means there are no hosts to report, but I'll add that to the error message.